crates/net/network/src/transactions/mod.rs

```diff
@@ -471,6 +471,74 @@ impl<Pool: TransactionPool, N: NetworkPrimitives> TransactionsManager<Pool, N> {
     }
 }
 
+/// decides which peers get transactions in a flexible and configurable way
+pub trait PeersPolicy {
+    /// Determines whether a transaction should be propagated to a given peer.
+    fn should_propagate(&self, peer: &PeerId) -> bool;
+
+    /// Called when a new peer is added to the network.
+    fn on_peer_added(&mut self, peer: PeerId);
+
+    /// Called when a peer is removed from the network.
+    fn on_peer_removed(&mut self, peer: &PeerId);
+}
+
+pub struct AllPeersPolicy;
+pub struct NoPeersPolicy;
+pub struct SubsetPeersPolicy {
+    allowed_peers: HashSet<PeerId>,
+}
+
+impl PeersPolicy for AllPeersPolicy {
+    fn should_propagate(&self, peer: &PeerId) -> bool {
+        true
+    }
+
+    fn on_peer_added(&mut self, peer: PeerId) {
+
+    }
+
+    fn on_peer_removed(&mut self, peer: &PeerId) {
+
+    }
```

Comment:
we can make this the default behaviour then we don't need to override all functions for each impl